### PR TITLE
[ENG-1612] Add brand-assets model + mirage setup

### PR DESCRIPTION
**Pull request by robustdj**
- Ticket: [ENG-1612]
- Feature flag: n/a

## Purpose

Adding `brand-assets` model along with mirage factories. This is also setting up the relationship between `registration-providers` and `brand-assets`

## Summary of Changes

* Add `brand-assets` model
* Add `brand-assets` factory
* Add `brand-assets` relationship on `registration-providers`

## QA Notes

[ENG-1612]: https://openscience.atlassian.net/browse/ENG-1612

TODO:

- [x] Add a test

### CODE REVIEW

File: app/models/brand-assets.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+const { attr } = DS;
+
+export default class BrandAssetsModel extends OsfModel {
+    @attr('string') primaryColor!: string;
+    @attr('string') secondaryColor!: string;
+    @attr('string') navbarLogoImage!: string;
+    @attr('string') heroLogoImage!: string;
+    @attr('string') heroBackgroundImage!: string;
+}
```
Comment (aaxelb) [inline app/models/brand-assets.ts]: you'll need to add the new model to the ember-data model registry ([more info on what/why in the ember-cli-typescript docs](https://ember-cli-typescript.com/docs/ts-guide/using-ts-effectively#string-keyed-lookups)), can copy how we do in [our other models](https://github.com/robustdj/ember-osf-web/blob/e6d4295e424feb494ecb9920211d1554d011b3e2/app/models/registration.ts#L137-L141)

File: mirage/factories/brand-assets.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    primaryColor: 'green',
+    secondaryColor: 'blue',
+    navbarLogoImage: 'http://somelogoimageurl',
+    heroLogoImage: 'http://somelogoimageurl',
+    heroBackgroundImage: 'http://herobackgroundimage',
+});
+
+declare module 'ember-cli-mirage/types/registries/model' {
+    export default interface MirageModelRegistry {
+        brandAssets: BrandAssets;
```
Comment (aaxelb) [inline mirage/factories/brand-assets.ts]:
```suggestion
        'brand-assets': BrandAssets;
```
for the type registries to work, the key needs to match the string used in string-based lookups

File: mirage/factories/brand-assets.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    } // eslint-disable-line semi
+}
+
+declare module 'ember-cli-mirage/types/registries/schema' {
+    export default interface MirageSchemaRegistry {
+        brandAssets: BrandAssets;
```
Comment (aaxelb) [inline mirage/factories/brand-assets.ts]:
```suggestion
        'brand-assets': BrandAssets;
```

**Comment by robustdj** [comment]
Still getting these 2 errors:

```
app/models/osf-model.ts:279:52 - error TS2345: Argument of type '"citation" | "comment-report" | "comment" | "file-version" | "file" | "license" | "subject" | "institutional-user" | "institution" | "schema-block" | "registration-schema" | ... 32 more ... | "user-registration"' is not assignable to parameter of type '"citation" | "comment-report" | "comment" | "file-version" | "file" | "license" | "subject" | "institutional-user" | "institution" | "schema-block" | "registration-schema" | ... 32 more ... | "file-contents"'.
  Type '"brand-assets"' is not assignable to type '"citation" | "comment-report" | "comment" | "file-version" | "file" | "license" | "subject" | "institutional-user" | "institution" | "schema-block" | "registration-schema" | ... 32 more ... | "file-contents"'.

279         const apiModelName = this.store.adapterFor(this.modelName).pathForType(this.modelName);
                                                       ~~~~~~~~~~~~~~

mirage/fixtures/registration-providers.ts:26:9 - error TS2740: Type 'ModelInstance<BrandAssetsModel>[]' is missing the following properties from type 'PromiseManyArray<BrandAssetsModel>': reload, createRecord, content, objectAtContent, and 30 more.

26         brandAssets: brandAssets(),
           ~~~~~~~~~~~

  app/models/registration-provider.ts:14:5
    14     brandAssets?: DS.PromiseManyArray<BrandAssetsModel>;
           ~~~~~~~~~~~
    The expected type comes from property 'brandAssets' which is declared here on type 'Partial<MirageRegistrationProvider>'

(node:12983) UnhandledPromiseRejectionWarning: Error: Typechecking failed
    at Class.postBuild (/Users/derrick/code/ember-osf-web/node_modules/ember-cli-typescript/js/addon.js:59:19)
    at process._tickCallback (internal/process/next_tick.js:68:7)
(node:12983) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 54)
```

**Comment by aaxelb** [comment]
ahh right, `brand-assets` needs to be in the adapter and serializer type registries, too -- we have a blueprint for this!
```
ember g osf-model brand-assets
```
^ should set up all the required pieces. sorry, forgot to mention earlier.

the type error in the fixture is a little trickier... will ponder a bit

File: mirage/fixtures/registration-providers.ts [hunk 4]
```diff
@@ -8,8 +8,13 @@ function randomAssets() {
     };
 }
 
+function brandAssets() {
+    return [server.create('brand-asset')];
+}
+
 interface MirageRegistrationProvider extends RegistrationProvider {
     licensesAcceptableIds: string[];
+    brandAssets: any;
```
Comment (robustdj) [inline mirage/fixtures/registration-providers.ts]: HACK!! I had to do this to fix a ts error. Not sure what the type should be here.

File: app/models/registration-provider.ts [hunk 5]
```diff
@@ -8,6 +9,9 @@ const { hasMany } = DS;
 export default class RegistrationProviderModel extends ProviderModel {
     @hasMany('registration', { inverse: 'provider' })
     registrations!: DS.PromiseManyArray<RegistrationModel>;
+
+    @hasMany('brand-asset', { inverse: null })
```
[the remark this answers is not in the document]
Comment (brianjgeiger) [reply app/models/registration-provider.ts]: The PR is still in review, I believe, so there should be time to change it without too much trouble. https://github.com/CenterForOpenScience/osf.io/pull/9315/

[the remark this answers is not in the document]
Comment (robustdj) [reply app/models/registration-provider.ts]: Another idea is `brand-asset-set`. We can always change things in the adapter too.

**Comment by aaxelb** [comment]
@robustdj looks like we're going with just `brand` for the model name instead of `brand-assets` or `brand-asset` -- sorry for the rename

File: mirage/fixtures/registration-providers.ts [hunk 6]
```diff
@@ -19,6 +20,7 @@ const registrationProviders: Array<Partial<MirageRegistrationProvider>> = [
         description: 'The open registries network',
         allowSubmissions: true,
         assets: randomAssets(),
+        brand: null,
```
Comment (aaxelb) [inline mirage/fixtures/registration-providers.ts]: per flowdock convo, this'll be set in another PR

**Comment by robustdj** [comment]
@aaxelb I forgot to rename the serializer from `brand-asset` to `brand`. Can you approve again and get this merged?

File: mirage/fixtures/registration-providers.ts [hunk 7]
```diff
@@ -10,6 +10,7 @@ function randomAssets() {
 
 interface MirageRegistrationProvider extends RegistrationProvider {
     licensesAcceptableIds: string[];
+    brand: any;
```
Comment (futa-ikeda) [inline mirage/fixtures/registration-providers.ts]: @aaxelb is this to be addressed in a later PR as well?

Comment (futa-ikeda) [reply mirage/fixtures/registration-providers.ts]: It may be prudent to type this as `Partial<Brand>` in here

Comment (robustdj) [reply mirage/fixtures/registration-providers.ts]: Can it be null as well?

Comment (aaxelb) [reply mirage/fixtures/registration-providers.ts]: since it's a type that's just used in this fixture (not exported), i'm ok letting this one `any` slip through for now... but yeah you're right, this should be addressed in that later PR

Comment (futa-ikeda) [reply mirage/fixtures/registration-providers.ts]: Good call, I believe it can be `null` as well! And if it's only in this mirage fixture, it shouldn't be too big of a deal 👍
